frontend/src/metabase/query_builder/components/DataSelector/DataSelector.jsx

```diff
@@ -330,13 +332,16 @@ export class UnconnectedDataSelector extends Component {
     }
 
     if (this.props.selectedDataBucketId === DATA_BUCKET.MODELS) {
-      this.showSavedEntityPicker();
+      this.showSavedEntityPicker({ entityType: "model" });
     }
 
     if (sourceId) {
       await this.props.fetchFields(sourceId);
       if (this.isSavedEntitySelected()) {
-        this.showSavedEntityPicker();
+        const id = getQuestionIdFromVirtualTableId(sourceId);
+        const question = this.props.metadata?.question(id);
```

Comment:
@kulyk suggested we don't get a question from the existing metadata in case it gets stale. He suggested using our entity loader for this. I haven't been able to find an elegant way to do this.

@ranquild does this solution seem acceptable, and/or would you also suggest fetching the question using the entity framework?

---
Before this change we'd always instantiate the "Saved question picker" (now renamed to the saved entity picker) for saved questions. Unless you were going through the "New > ..." flow, you could never see a model as a source.

With the introduction of new entities (models, metrics) this saved entity picker needs to differentiate between different entity types. So I needed to construct the question somehow to be able to use `question.type()`. Since I already have the question id, I thought the easiest way was to get it from the metadata.

Reply:
Using `metadata` here won't work. It might not be present in the metadata. 

Add to `mapDispatchToProps`:
```
fetchQuestion: selectedTableId => Qiestions.actions.fetch({ 
  id: getQuestionIdFromVirtualTableId(selectedTableId),
}),
```

then add to `mapStateToProps`
```
selectedQuestion: (state, { selectedTableId }) => Questions.selectors.getObject(state, { 
  entityId:  getQuestionIdFromVirtualTableId(selectedTableId),
}),
```

then here:
```
if (this.isSavedEntitySelected()) {
  await this.fetchQuestion(sourceId);
  this.showSavedEntityPicker({ entityType: this.selectedQuestion?.type() });
}
```

Reply:
Thank you.
Done in https://github.com/metabase/metabase/pull/39686/commits/513cb483b5de1507e20189f3d1a202c9456c38e2